### DISCUSSION TITLE
Localization - Language Config File Format

### DISCUSSION THREAD

**Discussion by bovirus** [radj307/volume-control#38] [Ideas & Feature Discussions]
@radj307 

In xx.loc.json file there is a string

  "LanguageName": {
    "**Italiano (Italian**)": "**Italiano**"

The first value ("italiano (Italian)" now is what appear in language selection drop down menu

The second value ("Italiano") I didn't udnerstand where is uaed.

In my vision should be work in different way

The first value should be always in English (ex. "italian") and should used for internal processing.

The sceond value should be the name that appear in drop down language file selection.
For me should be "Italiano (Italian)" where appear local name language plus english translation.
The iof someone made a wrong Italin selection can see also english value and identified it.

**Comment by radj307** [comment]
I agree that the value specified in `LanguageName` should be used for the display name, but since this is an issue with a dependency and not Volume Control itself, I can't fix it here; you should report this to the maintainer of the localization package we're using:  
https://github.com/codingseb/Localization

**Comment by bovirus** [comment]
@radj307

Please don't close this ticket until will be fixed in separate laocalization package.

Could you please open ia ticket in

https://github.com/codingseb/Localization

explaming in details the issue?

**Comment by radj307** [comment]
### Update

It is possible to resolve this by implementing `ILocalizationFileLoader` ourselves.  
This has some drawbacks:  
- **File Format Specific**  
  Both `json` & `yaml` require separate implementations.
- **Point of Failure**  
  Implementing this ourselves adds a point of failure; the file loaders provided by `CodingSeb.Localization` are already well-optimized, tested, and robust.

Since this is just a cosmetic issue, I don't think it's worth fixing due to the drawbacks.

**Comment by bovirus** [comment]
@radj307 

Ok. Thanks.
